src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloudTemplateIT.java

```diff
@@ -0,0 +1,136 @@
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.initCredentials;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.instanceConfiguration;
+import static com.google.jenkins.plugins.computeengine.integration.ITUtil.logs;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+import com.google.api.services.compute.model.Instance;
+import com.google.api.services.compute.model.InstanceTemplate;
+import com.google.common.collect.ImmutableMap;
+import com.google.jenkins.plugins.computeengine.ComputeEngineCloud;
+import com.google.jenkins.plugins.computeengine.client.ComputeClient;
+import hudson.model.labels.LabelAtom;
+import hudson.slaves.NodeProvisioner.PlannedNode;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+import java.util.logging.StreamHandler;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class ComputeEngineCloudTemplateIT {
+  private static Logger log = Logger.getLogger(ComputeEngineCloudTemplateIT.class.getName());
+
+  private static final String TEMPLATE =
+      format("projects/%s/global/instanceTemplates/test-template");
+  private static final String GOOGLE_LABEL_KEY = "test-label";
+  private static final String GOOGLE_LABEL_VALUE = "test-value";
+
+  @ClassRule public static Timeout timeout = new Timeout(5, TimeUnit.MINUTES);
+  @ClassRule public static JenkinsRule jenkinsRule = new JenkinsRule();
+
+  private static ByteArrayOutputStream logOutput = new ByteArrayOutputStream();
+  private static StreamHandler streamHandler = new StreamHandler(logOutput, new SimpleFormatter());
+  private static ComputeClient client;
+  private static Map<String, String> label = getLabel(ComputeEngineCloudTemplateIT.class);
+  private static ComputeEngineCloud cloud;
+  private static Instance instance;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    log.info("init");
+    initCredentials(jenkinsRule);
+    cloud = initCloud(jenkinsRule);
+    handleClassLogs(streamHandler, ComputeEngineCloud.class.getName());
+    client = initClient(jenkinsRule, label, log);
+    handleClassLogs(streamHandler, ComputeClient.class.getName());
+
+    cloud.addConfiguration(
+        instanceConfiguration(
+            DEB_JAVA_STARTUP_SCRIPT, NUM_EXECUTORS, LABEL, label, false, false, TEMPLATE));
+    InstanceTemplate instanceTemplate =
+        createTemplate(ImmutableMap.of(GOOGLE_LABEL_KEY, GOOGLE_LABEL_VALUE), TEMPLATE);
+    client.insertTemplate(cloud.projectId, instanceTemplate);
+    // Add a new node
+    Collection<PlannedNode> planned = cloud.provision(new LabelAtom(LABEL), 1);
```

Comment:
Rather than repeating this test throughout each init(), suggest creating a single test(s) which verify the provision code path and removing these from the init() methods.